### PR TITLE
Move variables to defaults to support more upgrade functionality

### PR DESCRIPTION
**Pull request by domenicbove**
# Description

Now `<component>_packages` list variables are customizable
Adds restart handlers to when packages get updated

Fixes # (https://github.com/confluentinc/cp-ansible/issues/591)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Ran through an upgrade, only using the install playbook with `deployment_strategy=rolling` and other variables

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any variable changes have been validated to be backwards compatible

### CODE REVIEW

File: roles/confluent.test/molecule/plaintext-rhel/molecule.yml [hunk 1]
```diff
@@ -158,6 +158,26 @@ provisioner:
         ksql_copy_files: "{{zookeeper_copy_files}}"
         control_center_copy_files: "{{zookeeper_copy_files}}"
 
+        deployment_strategy: rolling
```
Comment (domenicbove) [inline roles/confluent.test/molecule/plaintext-rhel/molecule.yml:161]: revert this file

**Comment by erikgb** [comment]
@domenicbove A question: Is the value of `confluent_server_enabled` really relevant when uninstalling an old/obsolete package?

**Comment by erikgb** [comment]
The uninstall of the deprecated package is quite repetitive code. Could it make sense to create a tasks-file in the `confluent.common` role - and supply the correct package list as a variable supplied to the `import_tasks` task for each component?

**Comment by domenicbove** [comment]
@erikgb Ya i dont really like the uninstall confluent-kafka-2.12 task. its only relevant for upgrading when confluent_server_enabled == false and you are going between a version below 6.0.0 to 6.0.0 or higher. 

Most of our customers have confluent_server_enabled = true...

And i suppose there is no harm in taking the when not confluent_server_enabled conditional away.

A lot of the tasks in a component role are repetitive :) I've thought about making a generic task file for everything, but haven't gone that path.

This could also be a good time to switch to the package module: https://docs.ansible.com/ansible/latest/collections/ansible/builtin/package_module.html

Would save some lines of code

**Comment by erikgb** [comment]
@domenicbove Are you able to upgrade from a version below 6.0 to version 6.0 or higher - **without** uninstalling that package? If yes, I would leave cleanup up to the user - and remove those tasks for now. It would of course be nice if cp-ansible could help the user in keeping a "tidy" system, but that would add significant maintainer overhead in cp-ansible. WDYT?

**Comment by domenicbove** [comment]
So again its when you are upgrading Apache Kafka, you need to remove the old package and install one of a new name. We also have `confluent-server` which is the confluent version of kafka, which does not have this problem. We install confluent-server by default.

Which package do you have installed?

I think we could phase wise roll this out for now, and remove the yum absent tasks.

**Comment by erikgb** [comment]
LGTM, but should we consider to move the main package variables as well. Makes little sense tweaking with those variables, but to be that seems more consistent.

Good choice to wait with the remove package tasks.

@domenicbove We have the `confluent-kafka-2.12` package installed in our 5.4 clusters.

**Comment by domenicbove** [comment]
@erikgb I think it's fine to force the user to use the `confluent_server_enabled` var to control `kafka_broker_main_package` but I don't feel that strongly either way.

So for your upgrade use case. you would need to stop the kafka service on a host, remove the `confluent-kafka-2.12` package, and then install on that host. Repeating on the rest of the hosts (with the controller last). Is that suitable?

**Comment by erikgb** [comment]
> So for your upgrade use case. you would need to stop the kafka service on a host, remove the `confluent-kafka-2.12` package, and then install on that host. Repeating on the rest of the hosts (with the controller last). Is that suitable?

We'll manage that upgrade process. 👍 But not sure how an upgrade could be included in the desired state and supported (for everyone). You want to it to be truly idempotent and avoid unnecessary stops/restarts. It's complicated..... 😄

**Review by xnulinu — APPROVED** [review]

**Comment by erikgb** [comment]
> So for your upgrade use case. you would need to stop the kafka service on a host, remove the `confluent-kafka-2.12` package, and then install on that host. Repeating on the rest of the hosts (with the controller last). Is that suitable?

@domenicbove We will probably end up doing the upgrade from 5.x to 6.x with the following additional tasks:

```yaml
- name: Get Package Facts
  package_facts:
    manager: auto

- block:
    - name: Stop Service
      ansible.builtin.systemd:
        name: "{{ service_name }}"
        state: stopped

    - name: Remove confluent-kafka-2.12 package
      ansible.builtin.package:
        name: confluent-kafka-2.12
        state: absent
  when: ansible_facts.packages['confluent-kafka-2.12'] is defined
```

So we'll definately manage, and thanks a lot again for your input/help! If you think this can help others upgrading Apache Kafka from 5.x to 6.x, I would be happy to submit a PR. WDYT? With regards to #591, I think that could be a step forward. BTW: I was inspired by this [upgrade tasks file](https://github.com/confluentinc/cp-ansible/blob/6.1.1-post/tasks/set_confluent_server_enabled.yml).

**Comment by domenicbove** [comment]
@erikgb You just reminded me- its probably best to stop the service anytime you are changing packages- this includes versions. Theres potential jars are getting replaced and the current running process could get in trouble.

So what you have is great but needs to also check versions (like the upgrade tasks do). As always, your PRs are welcome/appreciated!

**Pull request merged**